subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaCompile.java

```diff
@@ -57,6 +65,15 @@ public FileCollection getScalaClasspath() {
         return scalaClasspath;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @PathSensitive(PathSensitivity.RELATIVE)
```

Comment:
It seems like `NAME_ONLY` is the way to go. When changing the file name the actual byte code was different.